app/controllers/storage_controller.rb

```diff
@@ -121,78 +122,52 @@ def show(record = nil)
     @lastaction = "show"
   end
 
+  def specific_buttons(pressed)
+    if button_press_map[pressed]
+      send(button_press_map[pressed])
+    end
 
-  # handle buttons pressed on the button bar
-  def button
-    @edit = session[:edit]                                  # Restore @edit for adv search box
-    params[:display] = @display if %w(all_vms vms hosts).include?(@display) # Were we displaying vms or hosts
+    handle_host_power_buttons(pressed)
 
-    if params[:pressed].starts_with?("vm_", # Handle buttons from sub-items screen
-                                     "miq_template_",
-                                     "guest_",
-                                     "host_")
+    button_press_map.keys.include?(pressed)
+  end
 
-      scanhosts if params[:pressed] == "host_scan"
-      analyze_check_compliance_hosts if params[:pressed] == "host_analyze_check_compliance"
-      check_compliance_hosts if params[:pressed] == "host_check_compliance"
-      refreshhosts   if params[:pressed] == "host_refresh"
-      tag(Host) if params[:pressed] == "host_tag"
```

Comment:
this one is not working as it falls to the https://github.com/ManageIQ/manageiq/blob/master/app/controllers/mixins/generic_button_mixin.rb#L34 and the ```self.class``` here is Storage not Host